lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -223,37 +234,37 @@ public Map<String, Object> handleRequest(JourneyRequest event, Context context)
             if (configService.enabled(RESET_IDENTITY.getName())) {
                 return buildForceResetResponse();
             }
-            Gpg45Scores gpg45Scores = gpg45ProfileEvaluator.buildScore(credentials);
 
-            Optional<Gpg45Profile> matchedProfile =
-                    !userIdentityService.checkRequiresAdditionalEvidence(vcStoreItems)
-                            ? gpg45ProfileEvaluator.getFirstMatchingProfile(
-                                    gpg45Scores, CURRENT_ACCEPTED_GPG45_PROFILES)
-                            : Optional.empty();
-
-            // No profile match
-            if (matchedProfile.isEmpty()) {
-                return isF2FComplete
-                        ? buildF2FNoMatchResponse(auditEventUser)
-                        : buildNoMatchResponse(credentials, auditEventUser);
+            // Check for attained vot from vtr
+            var attainedVotAndProfile =
+                    getAttainedVotAndProfile(
+                            clientOAuthSessionItem.getVtr(),
+                            credentials,
+                            vcStoreItems,
+                            auditEventUser);
+
+            // vot achieved
+            if (attainedVotAndProfile.isPresent()) {
+                Vot attainedVot = attainedVotAndProfile.get().attainedVot();
+                ipvSessionItem.setVot(attainedVot.name());
+                ipvSessionService.updateIpvSession(ipvSessionItem);
+
+                return buildReuseResponse(attainedVot, auditEventUser);
             }
 
-            // Successful match
-            sendProfileMatchedAuditEvent(
-                    matchedProfile.get(), gpg45Scores, credentials, auditEventUser);
-
-            ipvSessionItem.setVot(VOT_P2);
-            ipvSessionService.updateIpvSession(ipvSessionItem);
+            // No profile match
+            return isF2FComplete
+                    ? buildF2FNoMatchResponse(auditEventUser)
+                    : buildNoMatchResponse(credentials, auditEventUser);
 
-            return buildProfileMatchedResponse(matchedProfile.get(), auditEventUser);
         } catch (HttpResponseExceptionWithErrorBody e) {
             LOGGER.error(LogHelper.buildErrorMessage(e.getErrorResponse().getMessage(), e));
             return new JourneyErrorResponse(
                             JOURNEY_ERROR_PATH, e.getResponseCode(), e.getErrorResponse())
                     .toObjectMap();
         } catch (CiRetrievalException e) {
             return buildErrorResponse(ErrorResponse.FAILED_TO_GET_STORED_CIS, e);
-        } catch (ParseException e) {
+        } catch (ParseException | JwtClaimsSetParseException e) {
```

Comment:
If you're catching this here, should it really be a runtime exception?

Reply:
It's a runtime exception as it's getting thrown in a stream, and you can't throw a checked exception from a stream. Unless you know a sneaky/better way?

Reply:
Bah 😞 

Yeah, the main alternative is just to avoid using a stream, which is almost as disappointing.

Reply:
Want me to remove the stream?

Reply:
Yeah, I think that's probably slightly more maintainable (but I'm not happy about it!)